### PR TITLE
feat(docs): add feature doc for FxA-54, phase 3 of connect another device

### LINKED ISSUE #246: Write the feature doc for deep-linking / pre-filled sign-in form

### PR DESCRIPTION
**Pull request by philbooth**
Fixes #246.

Attempts to boil down some of the extraneous information and conversation from the [original document](https://docs.google.com/document/d/1ObM7gZn0zt1NL4-3enBugZjJrxoD0ODthmuoaoEe7c0/edit), so that the end result is easier to understand when we come back to it in the future. Intentionally focused on the test plan and the metrics, since we presume those are what we'll need to come back to the most.

There's a mini-experiment of calling out the ubiquitous language in a dedicated section, not sure if that serves any purpose. The implementation plan has been kept down to a bare minimum.

@davismtl @ryanfeeley @mozilla/fxa-devs r? (no rush)

/cc @shane-tomlinson

### CODE REVIEW

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+3. Enter a valid phone number for an iOS device, submit.
+4. Open the SMS link on an iOS device.
+5. Safari will open and a dialog will ask `Open this page in "App Store"?` Touch `Open`.
+6. The Apple app store will open to the Firefox page. Click the install button.
+7. Firefox should install. When complete, click `Open`.
+8. Firefox will open to the Firefox for iOS firstrun page. Complete the firstrun flow.
+9. Firefox Accounts should open, email address from step 1 should be filled in.
+10. Sign in with the password from step 1.
+
+###### Sign up in Firefox desktop, send link to iOS, Firefox already installed
```
Comment (vbudhram) [inline features/FxA-54-connect-another-device-phase-3/README.md]: Assuming that using updated Firefox iOS (support deep and universal links into FxA), this should not redirect to App Store, but rather open app directly. Should be ok to remove step 6 here.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+within 2 days of creating an account:
+
+* https://sql.telemetry.mozilla.org/queries/5469
+
+These metrics and any others that are relevant
+will be tracked by charts on the following dashboard:
+
+* https://sql.telemetry.mozilla.org/dashboard/fxa-connect-another-device-phase-3
+
+## User flow
```
Comment (davismtl) [inline features/FxA-54-connect-another-device-phase-3/README.md]: You should include a brief description of the current flow so that the control is known and it is more clear how success is measured.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+   a mozilla or softvision address.
+2. In the same instance of Firefox desktop, open the verification link with
+   the `&signinCodes=true&forceExperimentGroup=treatment` query parameters.
+3. Enter a valid phone number for an Android device, submit.
+4. Open the SMS link on an Android device.
+5. Firefox should open to the firstrun flow. Complete the "firstrun flow".
+6. Firefox Accounts should open, email address from step 1 should be filled in.
+7. Sign in with the password from step 1.
+
+##### Reference Images
```
Comment (davismtl) [inline features/FxA-54-connect-another-device-phase-3/README.md]: If possible, include these images in-line above so that it is more clear. Else, link to the appropriate image at each step.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 4]
```diff
@@ -0,0 +1,367 @@
...
+###### Send SMS
+<img src="images/send_sms.png" width="250" alt="Send SMS"/>
+
+###### SMS received
+Image needed
+
+###### Install Firefox
+<img src="images/install_firefox_play_store_android.png" width="250" alt="Send SMS"/>
+
+###### Open Firefox
```
Comment (davismtl) [inline features/FxA-54-connect-another-device-phase-3/README.md]: @ryanfeeley should have new screenshots for when users open app from deeplink. The onboarding flow order on iOS should be changed from deeplinks to put FxA first (rather than last) since the user is completing a pairing flow.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 5]
```diff
@@ -0,0 +1,367 @@
...
+
+###### Sign up in Firefox desktop, send link to Android, Firefox already installed.
+
+1. Using Firefox desktop, sign in to Sync by creating a new Firefox Account using
+   a mozilla or softvision address.
+2. In the same instance of Firefox desktop, open the verification link with
+   the `&signinCodes=true&forceExperimentGroup=treatment` query parameters.
+3. Enter a valid phone number for an Android device, submit.
+4. Open the SMS link on an Android device.
+5. Firefox should open to the firstrun flow. Complete the "firstrun flow".
```
Comment (davismtl) [inline features/FxA-54-connect-another-device-phase-3/README.md]: I'm not sure why firstrun would open if app is already installed. Seems like we just need users to go FxA login form.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 6]
```diff
@@ -0,0 +1,367 @@
...
+1. Using Firefox desktop, sign in to Sync by creating a new Firefox Account using
+   a mozilla or softvision address.
+2. In the same instance of Firefox desktop, open the verification link with
+   the `&signinCodes=true&forceExperimentGroup=treatment` query parameters.
+3. Enter a valid phone number for an iOS device, submit.
+4. Open the SMS link on an iOS device. The link will be of the form `https://<server>/m/:signinCode`, write down the `signinCode`.
+5. Safari will open and a dialog will ask `Open this page in "App Store"?` Touch `Open`.
+6. The Apple app store will open to the Firefox page. Click the install button.
+7. Firefox should install. When complete, click `Open`.
+8. Firefox will open to the Firefox for iOS firstrun page. Click `Start Browsing`.
```
Comment (davismtl) [inline features/FxA-54-connect-another-device-phase-3/README.md]: This is where @ryanfeeley will have improved steps since we have a lot of drop-off from login.
If his improved steps are believed to be scope creep (not sure how well it was communicated), we can do an update after but it will be a step in the right direction to setup phase 4.

File: features/FxA-54-connect-another-device-phase-3/README.md (hunk 5 above)
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: This was written when the plan was still to add FxA after the firstrun flow completed. Will update.

File: features/FxA-54-connect-another-device-phase-3/README.md (hunk 3 above)
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: I think links are the way to go since each image pertains to multiple subsections. I tried putting the images inline to begin with and felt like I was staring at a giant wall of images and had a hard time putting text/images together.

File: features/FxA-54-connect-another-device-phase-3/README.md (hunk 5 above)
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: You are right, this needs to be updated. It wasn't clear how deep linking would work when I wrote this section. Neither Fx for iOS nor Android supported deep linking and at that time the feedback from mobile client UX folks was they wanted FxA to open afterwards.

File: features/FxA-54-connect-another-device-phase-3/README.md (hunk 4 above)
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: Good catch here too. The firstrun flow is out. The deep link causes Fx to open to `about:accounts?action=signin&signin=<signinCode>`

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 7]
```diff
@@ -0,0 +1,367 @@
...
+Reference images are below each OS section.
+
+signinCodes are disabled by default and can be enabled by opening the signup
+verification link with the `signinCodes=true` query parameter. Note, sending
+SMS is only available during the signup flow, signing into an existing account
+will not trigger the flow.
+
+#### Android
+
+##### Testing without support for deep links in Firefox for Android
```
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: Fx Beta channels support deep linking, and once https://github.com/mozilla/fxa-content-server/pull/5221 makes its way to prod, we'll be able to remove the "Testing without support for deep links" sections for both Firefox for Android and iOS. Instead of "testing without support", we can update the instructions to open the SMS link with the `channel=(beta|nightly)` to install different versions of Firefox.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 8]
```diff
@@ -0,0 +1,367 @@
...
+
+###### Sign up in Firefox desktop, send link to iOS, Firefox already installed
+
+1. Using Firefox desktop, sign in to Sync by creating a new Firefox Account using
+   a mozilla or softvision address.
+2. In the same instance of Firefox desktop, open the verification link with
+   the `&signinCodes=true&forceExperimentGroup=treatment` query parameters.
+3. Enter a valid phone number for an iOS device, submit.
+4. Open the SMS link on an iOS device.
+5. Firefox will open to the Firefox for iOS firstrun page. Complete the firstrun flow.
```
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: We can remove step 5, the firstrun flow is out.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 9]
```diff
@@ -0,0 +1,367 @@
...
+1. Using Firefox desktop, sign in to Sync by creating a new Firefox Account using
+   a mozilla or softvision address.
+2. In the same instance of Firefox desktop, open the verification link with
+   the `&signinCodes=true&forceExperimentGroup=treatment` query parameters.
+3. Enter a valid phone number for an iOS device, submit.
+4. Open the SMS link on an iOS device.
+5. Safari will open and a dialog will ask `Open this page in "App Store"?` Touch `Open`.
+6. The Apple app store will open to the Firefox page. Click the install button.
+7. Firefox should install. When complete, click `Open`.
+8. Firefox will open to the Firefox for iOS firstrun page. Complete the firstrun flow.
```
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: We can remove step 5, the firstrun flow is out.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 10]
```diff
@@ -0,0 +1,367 @@
...
+
+###### Firstrun
+Image needed
+
+##### Signin page with valid signinCode
+<img src="images/signin_valid_code_android.png" width="250" alt="Send SMS"/>
+
+#### iOS
+
+##### Testing without support for deep links in Firefox for iOS
```
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: Again, Fx beta release channel supports deep linking. We should be able to remove this section since https://github.com/mozilla/fxa-content-server/pull/5221 provides us ways of installing Beta instead of Release.

File: features/FxA-54-connect-another-device-phase-3/README.md [hunk 11]
```diff
@@ -0,0 +1,367 @@
...
+###### Sign up in Firefox desktop, send link to Android, Firefox not installed
+
+1. Using Firefox desktop, sign in to Sync by creating a new Firefox Account using
+   a mozilla or softvision address.
+2. In the same instance of Firefox desktop, open the verification link with
+   the `&signinCodes=true&forceExperimentGroup=treatment` query parameters.
+3. Enter a valid phone number for an Android device, submit.
+4. Open the SMS link on an Android device.
+5. Install Firefox from the Google Play store.
+6. Open Firefox, complete the "firstrun flow".
```
Comment (shane-tomlinson) [inline features/FxA-54-connect-another-device-phase-3/README.md]: Ditch step 6. Firstrun is out.

**Comment by philbooth** [comment]
> @philbooth - do you want me to make the requested updates to the "user flows" section?

No need, I've pushed a fix for that bit just now. You're welcome to crack on with the the test plan changes though.
